conn.go

```diff
@@ -358,6 +359,15 @@ func (hc *halfConn) decrypt(b *block) (ok bool, prefixLen int, alertValue alert)
 				hc.additionalData[11] = byte(n >> 8)
 				hc.additionalData[12] = byte(n)
 				additionalData = hc.additionalData[:]
+			} else {
+				if len(payload) > int((1<<14)+256) {
+					return false, 0, alertRecordOverflow
+				}
+				// Check AD header, see 5.2 of RFC8446
+				additionalData = make([]byte, 5)
+				additionalData[0] = byte(recordTypeApplicationData)
```

Comment:
It might be clearer to replace `recordTypeApplicationData` with `23` and `VersionTLS12` with `0x0303`, since the spec specifies these bytes concretely. Just a suggestion.

Here and below.

Reply:
ok, works for me